src/app/derive/derive.component.html

```diff
@@ -1,92 +1,99 @@
 <app-banner></app-banner>
 
-<div *ngIf="!globalVars.callbackInvalid">
-  <div class="container home-container" *ngIf="!deleteKey">
-    <div class="font-weight-bold fs-30px mb-20px" *ngIf="!globalVars.callback">
-      Grant signing access to {{ globalVars.hostname }}
-    </div>
-    <div class="font-weight-bold fs-30px mb-20px" *ngIf="globalVars.callback">
+<div
+  *ngIf="
+    !globalVars.callbackInvalid && !deleteKey && validationErrors === false
+  "
+  class="container home-container text-center d-flex flex-column"
+>
+  <div>
+    <div class="title-text">
       Grant signing access
+      <span *ngIf="!globalVars.callback">to {{ globalVars.hostname }}</span>
     </div>
-    <div
-      class="font-weight-bold fs-18px mb-20px"
-      style="overflow: hidden; overflow-wrap: anywhere; word-break: break-all;"
-      *ngIf="publicKeyBase58Check"
-    >
-      Public Key: {{ publicKeyBase58Check }}
+
+    <div class="d-flex flex-column" *ngIf="hasUsers && !isSingleAccount">
+      <ul
+        class="list-group mt-7px mb-30px saved-seeds-list saved-seeds-scroll"
+        style="max-height: 700px"
+      >
+        <span class="saved-seeds-header d-flex align-items-center"
+          ><span>Select an account</span></span
+        >
+        <li
+          *ngFor="let item of allUsers | keyvalue; let i = index"
+          class="list-group-item list-group-item-action cursor-pointer saved-seed"
+          [ngClass]="{ selectedKey: publicKeyBase58Check === item.key }"
+          (click)="publicKeyBase58Check = item.key"
+        >
+          <div class="w-100 d-flex justify-content-between">
+            <div *ngIf="!item.value.username" class="text-left">
+              <div class="my-1">
+                <img
+                  class="avatar mr-15px"
+                  src="assets/placeholder-account-image.png"
+                />
+                {{ truncatePublicKey(item.key) }}
+              </div>
+            </div>
+            <div *ngIf="item.value.username" class="d-flex align-items-center">
+              <div class="avatar mr-15px" [appAvatar]="item.key"></div>
+              <div class="">{{ item.value.username }}</div>
+            </div>
+          </div>
+        </li>
+      </ul>
     </div>
     <div
-      class="font-weight-bold fs-18px mb-20px"
-      style="overflow: hidden; overflow-wrap: anywhere; word-break: break-all;"
+      class="fs-18px pt-15px pb-5px"
+      style="overflow: hidden; overflow-wrap: anywhere; word-break: break-all"
       *ngIf="derivedPublicKeyBase58Check"
     >
-      Derived Public Key: {{ derivedPublicKeyBase58Check }}
```

Comment:
where are we displaying the derived public key? This is necessary for apps that want to specify a derived public key when asking the user to approve a derived key instead of generating a new one.

Reply:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/98359380/184460688-a91949ae-f15c-42e3-96b7-6d979f02f959.png">

Reply:
<img width="711" alt="image" src="https://user-images.githubusercontent.com/98359380/184460741-21db2d0f-21d2-4985-90c2-5149a0867af2.png">